src/network/sync_item_message.cpp

```diff
@@ -60,14 +60,12 @@ void SyncItemMessageHandlerSubSystem::Execute(Message const& message)
     {
         return;
     }
-    Opt<Item> item = inv->GetItem( msg.mItemID );
-    if( !item.IsValid() )
-    {
-        return;
-    }
+    inv->DropItem( msg.mItemID );
     std::istringstream iss( msg.mData );
     eos::portable_iarchive ia(iss);
-    ia >> *item;
+    Opt<Item> item;
+    ia >> item;
+    inv->AddItem(std::unique_ptr<Item>(item.Get()));
```

Comment:
store id, don't use item after its ownership was transferred (might no longer be valid based on the interface)
or use shared_ptr